src/thumbnailer.rs

```diff
@@ -195,87 +201,91 @@ impl Thumbnailer {
     }
 
     fn save_thumbnail_to_temp(thumbnailer: Thumbnailer) -> Result<Thumbnailer, String> {
-        &thumbnailer
-            .thumbnail
-            .as_ref()
-            .unwrap()
-            .save_with_format(
-                thumbnailer.temp_path.to_str().unwrap(),
-                image::ImageFormat::PNG,
-            )
-            .map_err(|e| {
-                error!("{}", e);
-                "Failed to save thumbnail".to_owned()
-            });
-        Ok(thumbnailer)
-    }
-
-    fn update_metadata(thumbnailer: Thumbnailer) -> Result<Thumbnailer, String> {
-        let mut chunks = {
-            let mut input = std::fs::File::open(&thumbnailer.temp_path)
-                .map_err(|_e| "Failed to open thumbnailer in temporary dir".to_owned())?;
-            crate::png::Png::decode(&mut input).map_err(|_e| "Failed decoding chunks".to_owned())?
+        let temp_path = format!("{}.tmp", thumbnailer.destination_path.to_str().unwrap());
+        let output = std::fs::OpenOptions::new()
+            .write(true)
+            .truncate(true)
+            .create(true)
+            .open(&temp_path)
+            .map_err(|e| format!("Failed to open thumbnailer in temporary dir: {}", e))?;
+
+        let thumbnail = thumbnailer.thumbnail.as_ref().unwrap();
+        let (ct, bits) = match thumbnail.color() {
```

Comment:
Do we really need to get low level and query the original image format?

Reply:
Yeah, the image-to-png type mapping is a little ugly.

To write the png in one pass, we need to get a hold of the PNG encoder to write the metadata chunks. This rules out image's save_with_format, as it doesn't expose the encoder. Instead, I just [took this image-to-png type mapping straight from the image lib](https://docs.rs/image/0.23.6/src/image/png.rs.html#508).